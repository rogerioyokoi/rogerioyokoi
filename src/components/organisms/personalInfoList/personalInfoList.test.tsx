import { personalInfoData } from '@/assets/data/meta.const';
import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { axe } from 'vitest-axe';
import PersoanlInfoList from './personalInfoList';

describe('Components > Organisms > PersoanlInfoList', () => {
  describe('Snapshot', () => {
    it('Should match snapshot', () => {
      const { container } = render(<PersoanlInfoList />);

      expect(container).toMatchSnapshot();
    });
  });

  describe('Accessibility Tests (A11y)', () => {
    it('Should not have accessibility violations', async () => {
      const { container } = render(<PersoanlInfoList />);

      expect(await axe(container)).toHaveNoViolations();
    });
  });

  describe('Rendering', () => {
    it('Should render list correctly', () => {
      const { getByTestId } = render(<PersoanlInfoList />);

      const iElmList = getByTestId('personal-info-list');

      expect(iElmList).toBeInTheDocument();
    });

    it('Should render list items correctly', () => {
      const { queryAllByTestId } = render(<PersoanlInfoList />);

      const iElmListItem = queryAllByTestId('personal-info-list-item');

      expect(iElmListItem.length).toBe(personalInfoData.length);
    });
  });

  describe('CSS Classes', () => {
    it('Should apply correct classes to list', () => {
      const { getByTestId } = render(<PersoanlInfoList />);
      const headingElement = getByTestId('personal-info-list');

      expect(headingElement).toHaveClass('list-none grid grid-cols-2 gap-4');
    });

    it('Should apply correct classes to key on list item', () => {
      const { queryAllByTestId } = render(<PersoanlInfoList />);

      const iElmListItemKey = queryAllByTestId('personal-info-list-item-key')[0];

      expect(iElmListItemKey).toHaveClass('font-bold');
    });

    it('Should apply correct classes to value on list item', () => {
      const { queryAllByTestId } = render(<PersoanlInfoList />);

      const iElmListItemValue = queryAllByTestId('personal-info-list-item-value')[0];

      expect(iElmListItemValue).toHaveClass('ml-2 font-light');
    });
  });
});
