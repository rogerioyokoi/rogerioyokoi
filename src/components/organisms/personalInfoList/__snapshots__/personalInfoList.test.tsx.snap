// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components > Organisms > PersoanlInfoList > Snapshot > Should match snapshot 1`] = `
<div>
  <ul
    class="list-none grid grid-cols-2 gap-4"
    data-testid="personal-info-list"
  >
    <li
      data-testid="personal-info-list-item"
    >
      <span
        class="font-bold"
        data-testid="personal-info-list-item-key"
      >
        Nome
        :
      </span>
      <span
        class="ml-2 font-light"
        data-testid="personal-info-list-item-value"
      >
        Rogério Yokoi
      </span>
    </li>
    <li
      data-testid="personal-info-list-item"
    >
      <span
        class="font-bold"
        data-testid="personal-info-list-item-key"
      >
        Idade
        :
      </span>
      <span
        class="ml-2 font-light"
        data-testid="personal-info-list-item-value"
      >
        40 anos
      </span>
    </li>
    <li
      data-testid="personal-info-list-item"
    >
      <span
        class="font-bold"
        data-testid="personal-info-list-item-key"
      >
        Nacionalidade
        :
      </span>
      <span
        class="ml-2 font-light"
        data-testid="personal-info-list-item-value"
      >
        Brasileira
      </span>
    </li>
    <li
      data-testid="personal-info-list-item"
    >
      <span
        class="font-bold"
        data-testid="personal-info-list-item-key"
      >
        Residência
        :
      </span>
      <span
        class="ml-2 font-light"
        data-testid="personal-info-list-item-value"
      >
        Uberaba / MG
      </span>
    </li>
    <li
      data-testid="personal-info-list-item"
    >
      <span
        class="font-bold"
        data-testid="personal-info-list-item-key"
      >
        Idiomas
        :
      </span>
      <span
        class="ml-2 font-light"
        data-testid="personal-info-list-item-value"
      >
        Português (Nativo), Inglês (Técnico)
      </span>
    </li>
  </ul>
</div>
`;
